desktop/main.ts

```diff
@@ -32,11 +34,68 @@ const MAC_PERMISSION_STATUSES = {
     NOT_DETERMINED: 'not determined',
 } as const;
 
-const LOCATION_PERMISSION_STATES = {
-    GRANTED: 'granted',
-    DENIED: 'denied',
-    PROMPT: 'prompt',
-} as const;
+type MacPermissionsModule = {
+    getAuthStatus?: (authType: AuthType) => PermissionType | typeof MAC_PERMISSION_STATUSES.NOT_DETERMINED;
+};
+
+type MacGetAuthStatus = MacPermissionsModule['getAuthStatus'];
+
+let macGetAuthStatusPromise: Promise<MacGetAuthStatus | undefined> | undefined;
+let hasLoggedMacPermissionsWarning = false;
+
+const logMacPermissionsWarningOnce = (message: string, error?: unknown) => {
+    if (hasLoggedMacPermissionsWarning) {
+        return;
+    }
+
+    if (error instanceof Error) {
+        log.warn(message, error.message);
+    } else if (typeof error === 'string') {
+        log.warn(message, error);
+    } else {
+        log.warn(message);
+    }
+
+    hasLoggedMacPermissionsWarning = true;
+};
+
+const loadMacGetAuthStatus = async (): Promise<MacGetAuthStatus | undefined> => {
+    if (!macGetAuthStatusPromise) {
+        if (process.platform !== 'darwin') {
+            macGetAuthStatusPromise = Promise.resolve<MacGetAuthStatus | undefined>(undefined);
+        } else {
+            macGetAuthStatusPromise = import('node-mac-permissions')
```

Comment:
You can use `await` here


```suggestion
            macGetAuthStatusPromise = await import('node-mac-permissions');
```